lib/ansible/executor/module_common.py

```diff
@@ -1334,13 +1350,21 @@ def get_action_args_with_defaults(action, args, defaults, templar):
         module_defaults = templar.template(module_defaults)
 
         # deal with configured group defaults first
-        if action in C.config.module_defaults_groups:
-            for group in C.config.module_defaults_groups.get(action, []):
-                tmp_args.update((module_defaults.get('group/{0}'.format(group)) or {}).copy())
+        for group_name in collection_groups:
+            if 'group/%s' % group_name not in module_defaults:
+                continue
+
+            for collection_name in collection_groups[group_name]:
+                collection_routing = _get_collection_metadata(collection_name).get('action_groups', {})
```

Comment:
This doesn't currently appear resilient to missing collection(s) in the group's list (eg, you only have `amazon.aws` and not `community.aws` available).